acceptable/openapi.py

```diff
@@ -22,32 +23,147 @@ def _to_dict(source: Any):
         return source
 
 
+def tidy_string(untidy: str):
+    tidy = str(untidy).replace("\n", " ")
+    while "  " in tidy:
+        tidy = tidy.replace("  ", " ")
+    return tidy.strip()
+
+
+def get_head_of_single_item_list(item_list, item_name):
```

Comment:
Hm this function looks too short and specific, imho. Instead of:
```py
method = get_head_of_single_item_list(endpoint.methods, "method")
```
I would probably just do this inline in the caller:
```py
try:
    [method] = endpoint.methods
except TypeError:
    logging.warning("something... ")
    method = None
```